# Lending Club Case Study

## Initial Setup

Import required libraries for Exploratory Data Analysis. 

In [228]:
import numpy as np, pandas as pd, matplotlib.pyplot as plt, seaborn as sns

Load the loan dataset

In [229]:
loan_data = pd.read_csv('./loan.csv', low_memory=False)

In [230]:
loan_data.describe()

id     member_id     loan_amnt   funded_amnt  \
count  3.971700e+04  3.971700e+04  39717.000000  39717.000000   
mean   6.831319e+05  8.504636e+05  11219.443815  10947.713196   
std    2.106941e+05  2.656783e+05   7456.670694   7187.238670   
min    5.473400e+04  7.069900e+04    500.000000    500.000000   
25%    5.162210e+05  6.667800e+05   5500.000000   5400.000000   
50%    6.656650e+05  8.508120e+05  10000.000000   9600.000000   
75%    8.377550e+05  1.047339e+06  15000.000000  15000.000000   
max    1.077501e+06  1.314167e+06  35000.000000  35000.000000   

       funded_amnt_inv   installment    annual_inc           dti  \
count     39717.000000  39717.000000  3.971700e+04  39717.000000   
mean      10397.448868    324.561922  6.896893e+04     13.315130   
std        7128.450439    208.874874  6.379377e+04      6.678594   
min           0.000000     15.690000  4.000000e+03      0.000000   
25%        5000.000000    167.020000  4.040400e+04      8.170000   
50%        8975.000000    280.220000  5.900000e+04     13.400000   
75%       14400.000000    430.780000  8.230000e+04     18.600000   
max       35000.000000   1305.190000  6.000000e+06     29.990000   

        delinq_2yrs  inq_last_6mths  ...  num_tl_90g_dpd_24m  \
count  39717.000000    39717.000000  ...                 0.0   
mean       0.146512        0.869200  ...                 NaN   
std        0.491812        1.070219  ...                 NaN   
min        0.000000        0.000000  ...                 NaN   
25%        0.000000        0.000000  ...                 NaN   
50%        0.000000        1.000000  ...                 NaN   
75%        0.000000        1.000000  ...                 NaN   
max       11.000000        8.000000  ...                 NaN   

       num_tl_op_past_12m  pct_tl_nvr_dlq  percent_bc_gt_75  \
count                 0.0             0.0               0.0   
mean                  NaN             NaN               NaN   
std                   NaN             NaN               NaN   
min                   NaN             NaN               NaN   
25%                   NaN             NaN               NaN   
50%                   NaN             NaN               NaN   
75%                   NaN             NaN               NaN   
max                   NaN             NaN               NaN   

       pub_rec_bankruptcies  tax_liens  tot_hi_cred_lim  total_bal_ex_mort  \
count          39020.000000    39678.0              0.0                0.0   
mean               0.043260        0.0              NaN                NaN   
std                0.204324        0.0              NaN                NaN   
min                0.000000        0.0              NaN                NaN   
25%                0.000000        0.0              NaN                NaN   
50%                0.000000        0.0              NaN                NaN   
75%                0.000000        0.0              NaN                NaN   
max                2.000000        0.0              NaN                NaN   

       total_bc_limit  total_il_high_credit_limit  
count             0.0                         0.0  
mean              NaN                         NaN  
std               NaN                         NaN  
min               NaN                         NaN  
25%               NaN                         NaN  
50%               NaN                         NaN  
75%               NaN                         NaN  
max               NaN                         NaN  

[8 rows x 87 columns]

In [231]:
# Get a preview
loan_data.head()

id  member_id  loan_amnt  funded_amnt  funded_amnt_inv        term  \
0  1077501    1296599       5000         5000           4975.0   36 months   
1  1077430    1314167       2500         2500           2500.0   60 months   
2  1077175    1313524       2400         2400           2400.0   36 months   
3  1076863    1277178      10000        10000          10000.0   36 months   
4  1075358    1311748       3000         3000           3000.0   60 months   

  int_rate  installment grade sub_grade  ... num_tl_90g_dpd_24m  \
0   10.65%       162.87     B        B2  ...                NaN   
1   15.27%        59.83     C        C4  ...                NaN   
2   15.96%        84.33     C        C5  ...                NaN   
3   13.49%       339.31     C        C1  ...                NaN   
4   12.69%        67.79     B        B5  ...                NaN   

  num_tl_op_past_12m pct_tl_nvr_dlq  percent_bc_gt_75 pub_rec_bankruptcies  \
0                NaN            NaN               NaN                  0.0   
1                NaN            NaN               NaN                  0.0   
2                NaN            NaN               NaN                  0.0   
3                NaN            NaN               NaN                  0.0   
4                NaN            NaN               NaN                  0.0   

  tax_liens tot_hi_cred_lim total_bal_ex_mort total_bc_limit  \
0       0.0             NaN               NaN            NaN   
1       0.0             NaN               NaN            NaN   
2       0.0             NaN               NaN            NaN   
3       0.0             NaN               NaN            NaN   
4       0.0             NaN               NaN            NaN   

  total_il_high_credit_limit  
0                        NaN  
1                        NaN  
2                        NaN  
3                        NaN  
4                        NaN  

[5 rows x 111 columns]

In [232]:
# Drop columns that are completely empty
loan_data.dropna(axis='columns', how='all', inplace=True)

# Verify column count
loan_data.head(5)

id  member_id  loan_amnt  funded_amnt  funded_amnt_inv        term  \
0  1077501    1296599       5000         5000           4975.0   36 months   
1  1077430    1314167       2500         2500           2500.0   60 months   
2  1077175    1313524       2400         2400           2400.0   36 months   
3  1076863    1277178      10000        10000          10000.0   36 months   
4  1075358    1311748       3000         3000           3000.0   60 months   

  int_rate  installment grade sub_grade  ... next_pymnt_d last_credit_pull_d  \
0   10.65%       162.87     B        B2  ...          NaN             May-16   
1   15.27%        59.83     C        C4  ...          NaN             Sep-13   
2   15.96%        84.33     C        C5  ...          NaN             May-16   
3   13.49%       339.31     C        C1  ...          NaN             Apr-16   
4   12.69%        67.79     B        B5  ...       Jun-16             May-16   

  collections_12_mths_ex_med  policy_code application_type acc_now_delinq  \
0                        0.0            1       INDIVIDUAL              0   
1                        0.0            1       INDIVIDUAL              0   
2                        0.0            1       INDIVIDUAL              0   
3                        0.0            1       INDIVIDUAL              0   
4                        0.0            1       INDIVIDUAL              0   

  chargeoff_within_12_mths delinq_amnt pub_rec_bankruptcies tax_liens  
0                      0.0           0                  0.0       0.0  
1                      0.0           0                  0.0       0.0  
2                      0.0           0                  0.0       0.0  
3                      0.0           0                  0.0       0.0  
4                      0.0           0                  0.0       0.0  

[5 rows x 57 columns]

111 - 57 = __54 columns__ are dropped as they did not have any values and hence not useful for analysis.

In [233]:
# Inspect datatype of each column to see if any of the datatypes don't make sense
loan_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 39717 entries, 0 to 39716
Data columns (total 57 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   id                          39717 non-null  int64  
 1   member_id                   39717 non-null  int64  
 2   loan_amnt                   39717 non-null  int64  
 3   funded_amnt                 39717 non-null  int64  
 4   funded_amnt_inv             39717 non-null  float64
 5   term                        39717 non-null  object 
 6   int_rate                    39717 non-null  object 
 7   installment                 39717 non-null  float64
 8   grade                       39717 non-null  object 
 9   sub_grade                   39717 non-null  object 
 10  emp_title                   37258 non-null  object 
 11  emp_length                  38642 non-null  object 
 12  home_ownership              39717 non-null  object 
 13  annual_inc                  397

A lot of columns have `object` data type. Let's look at these columns in isolation to better understand if `object` is indeed the right data type for these columns. 

In [234]:
loan_data.select_dtypes('object').head(5)

term int_rate grade sub_grade                 emp_title emp_length  \
0   36 months   10.65%     B        B2                       NaN  10+ years   
1   60 months   15.27%     C        C4                     Ryder   < 1 year   
2   36 months   15.96%     C        C5                       NaN  10+ years   
3   36 months   13.49%     C        C1       AIR RESOURCES BOARD  10+ years   
4   60 months   12.69%     B        B5  University Medical Group     1 year   

  home_ownership verification_status issue_d  loan_status  ...  \
0           RENT            Verified  Dec-11   Fully Paid  ...   
1           RENT     Source Verified  Dec-11  Charged Off  ...   
2           RENT        Not Verified  Dec-11   Fully Paid  ...   
3           RENT     Source Verified  Dec-11   Fully Paid  ...   
4           RENT     Source Verified  Dec-11      Current  ...   

                  title zip_code addr_state earliest_cr_line revol_util  \
0              Computer    860xx         AZ           Jan-85     83.70%   
1                  bike    309xx         GA           Apr-99      9.40%   
2  real estate business    606xx         IL           Nov-01     98.50%   
3              personel    917xx         CA           Feb-96        21%   
4              Personal    972xx         OR           Jan-96     53.90%   

  initial_list_status last_pymnt_d next_pymnt_d last_credit_pull_d  \
0                   f       Jan-15          NaN             May-16   
1                   f       Apr-13          NaN             Sep-13   
2                   f       Jun-14          NaN             May-16   
3                   f       Jan-15          NaN             Apr-16   
4                   f       May-16       Jun-16             May-16   

  application_type  
0       INDIVIDUAL  
1       INDIVIDUAL  
2       INDIVIDUAL  
3       INDIVIDUAL  
4       INDIVIDUAL  

[5 rows x 24 columns]

As per the data dictionary, the column definition for `term` is "The number of payments on the loan. Values are in months and can be either `36` or `60`". As it is, the term column is not suitable for analysis as it contains strings. Let's convert it into a integer column. 

In [235]:
# Check if the term column has any empty values
loan_data.term.isna().sum()

0

The term column does not contain any null values. Let's convert the data type now. 

In [236]:
loan_data.term = loan_data.term.apply(lambda x: int(x.strip().split(" ")[0]))

In [237]:
# Verify that the data type is correct
loan_data.term.head(5)

0    36
1    60
2    36
3    36
4    60
Name: term, dtype: int64

In [238]:
loan_data.select_dtypes('object')

int_rate grade sub_grade                   emp_title emp_length  \
0       10.65%     B        B2                         NaN  10+ years   
1       15.27%     C        C4                       Ryder   < 1 year   
2       15.96%     C        C5                         NaN  10+ years   
3       13.49%     C        C1         AIR RESOURCES BOARD  10+ years   
4       12.69%     B        B5    University Medical Group     1 year   
...        ...   ...       ...                         ...        ...   
39712    8.07%     A        A4             FiSite Research    4 years   
39713   10.28%     C        C1  Squarewave Solutions, Ltd.    3 years   
39714    8.07%     A        A4                         NaN   < 1 year   
39715    7.43%     A        A2                         NaN   < 1 year   
39716   13.75%     E        E2            Evergreen Center   < 1 year   

      home_ownership verification_status issue_d  loan_status pymnt_plan  ...  \
0               RENT            Verified  Dec-11   Fully Paid          n  ...   
1               RENT     Source Verified  Dec-11  Charged Off          n  ...   
2               RENT        Not Verified  Dec-11   Fully Paid          n  ...   
3               RENT     Source Verified  Dec-11   Fully Paid          n  ...   
4               RENT     Source Verified  Dec-11      Current          n  ...   
...              ...                 ...     ...          ...        ...  ...   
39712       MORTGAGE        Not Verified  Jul-07   Fully Paid          n  ...   
39713           RENT        Not Verified  Jul-07   Fully Paid          n  ...   
39714       MORTGAGE        Not Verified  Jul-07   Fully Paid          n  ...   
39715       MORTGAGE        Not Verified  Jul-07   Fully Paid          n  ...   
39716            OWN        Not Verified  Jun-07   Fully Paid          n  ...   

                           title zip_code addr_state earliest_cr_line  \
0                       Computer    860xx         AZ           Jan-85   
1                           bike    309xx         GA           Apr-99   
2           real estate business    606xx         IL           Nov-01   
3                       personel    917xx         CA           Feb-96   
4                       Personal    972xx         OR           Jan-96   
...                          ...      ...        ...              ...   
39712           Home Improvement    802xx         CO           Nov-90   
39713  Retiring credit card debt    274xx         NC           Dec-86   
39714     MBA Loan Consolidation    017xx         MA           Oct-98   
39715                   JAL Loan    208xx         MD           Nov-88   
39716         Consolidation Loan    027xx         MA           Oct-03   

      revol_util initial_list_status last_pymnt_d next_pymnt_d  \
0         83.70%                   f       Jan-15          NaN   
1          9.40%                   f       Apr-13          NaN   
2         98.50%                   f       Jun-14          NaN   
3            21%                   f       Jan-15          NaN   
4         53.90%                   f       May-16       Jun-16   
...          ...                 ...          ...          ...   
39712     13.10%                   f       Jul-10          NaN   
39713     26.90%                   f       Jul-10          NaN   
39714     19.40%                   f       Apr-08          NaN   
39715      0.70%                   f       Jan-08          NaN   
39716     51.50%                   f       Jun-10          NaN   

      last_credit_pull_d application_type  
0                 May-16       INDIVIDUAL  
1                 Sep-13       INDIVIDUAL  
2                 May-16       INDIVIDUAL  
3                 Apr-16       INDIVIDUAL  
4                 May-16       INDIVIDUAL  
...                  ...              ...  
39712             Jun-10       INDIVIDUAL  
39713             Jul-10       INDIVIDUAL  
39714             Jun-07       INDIVIDUAL  
39715             Jun-07       INDIVIDUAL  
39

It can also be noticed that the `int_rate` column which denotes the interest rate of the loan is a string. Let's change that to a float. 

In [239]:
# Let's verify if there are any empty values for interest rate
loan_data.int_rate.isna().sum()

0

In [240]:
# Since there aren't any empty values, let's proceed with converting the data type
loan_data.int_rate = loan_data.int_rate.apply(lambda x: float(x[:-1]))

In [241]:
# Verify that the data type is correct
loan_data.int_rate.head(5)

0    10.65
1    15.27
2    15.96
3    13.49
4    12.69
Name: int_rate, dtype: float64

In [242]:
loan_data.select_dtypes('object')

grade sub_grade                   emp_title emp_length home_ownership  \
0         B        B2                         NaN  10+ years           RENT   
1         C        C4                       Ryder   < 1 year           RENT   
2         C        C5                         NaN  10+ years           RENT   
3         C        C1         AIR RESOURCES BOARD  10+ years           RENT   
4         B        B5    University Medical Group     1 year           RENT   
...     ...       ...                         ...        ...            ...   
39712     A        A4             FiSite Research    4 years       MORTGAGE   
39713     C        C1  Squarewave Solutions, Ltd.    3 years           RENT   
39714     A        A4                         NaN   < 1 year       MORTGAGE   
39715     A        A2                         NaN   < 1 year       MORTGAGE   
39716     E        E2            Evergreen Center   < 1 year            OWN   

      verification_status issue_d  loan_status pymnt_plan  \
0                Verified  Dec-11   Fully Paid          n   
1         Source Verified  Dec-11  Charged Off          n   
2            Not Verified  Dec-11   Fully Paid          n   
3         Source Verified  Dec-11   Fully Paid          n   
4         Source Verified  Dec-11      Current          n   
...                   ...     ...          ...        ...   
39712        Not Verified  Jul-07   Fully Paid          n   
39713        Not Verified  Jul-07   Fully Paid          n   
39714        Not Verified  Jul-07   Fully Paid          n   
39715        Not Verified  Jul-07   Fully Paid          n   
39716        Not Verified  Jun-07   Fully Paid          n   

                                                     url  ...  \
0      https://lendingclub.com/browse/loanDetail.acti...  ...   
1      https://lendingclub.com/browse/loanDetail.acti...  ...   
2      https://lendingclub.com/browse/loanDetail.acti...  ...   
3      https://lendingclub.com/browse/loanDetail.acti...  ...   
4      https://lendingclub.com/browse/loanDetail.acti...  ...   
...                                                  ...  ...   
39712  https://lendingclub.com/browse/loanDetail.acti...  ...   
39713  https://lendingclub.com/browse/loanDetail.acti...  ...   
39714  https://lendingclub.com/browse/loanDetail.acti...  ...   
39715  https://lendingclub.com/browse/loanDetail.acti...  ...   
39716  https://lendingclub.com/browse/loanDetail.acti...  ...   

                           title zip_code addr_state earliest_cr_line  \
0                       Computer    860xx         AZ           Jan-85   
1                           bike    309xx         GA           Apr-99   
2           real estate business    606xx         IL           Nov-01   
3                       personel    917xx         CA           Feb-96   
4                       Personal    972xx         OR           Jan-96   
...                          ...      ...        ...              ...   
39712           Home Improvement    802xx         CO           Nov-90   
39713  Retiring credit card debt    274xx         NC           Dec-86   
39714     MBA Loan Consolidation    017xx         MA           Oct-98   
39715                   JAL Loan    208xx         MD           Nov-88   
39716         Consolidation Loan    027xx         MA           Oct-03   

      revol_util initial_list_status last_pymnt_d next_pymnt_d  \
0         83.70%                   f       Jan-15          NaN   
1          9.40%                   f       Apr-13          NaN   
2         98.50%                   f       Jun-14          NaN   
3            21%                   f       Jan-15          NaN   
4         53.90%                   f       May-16       Jun-16   
...          ...                 ...          ...          ...   
39712     13.10%                   f       Jul-10          NaN   
39713     26.90%                   f       Jul-10          NaN   
39714     19.40%                   f       Apr-08          NaN   
39715 

In [243]:
issue_d = pd.to_datetime(loan_data.issue_d, format="%b-%y").dt
loan_data['issue_month'] = issue_d.month
loan_data['issue_year'] = issue_d.year

In [244]:
loan_data.loc[:, ['issue_month', 'issue_year']]

issue_month  issue_year
0               12        2011
1               12        2011
2               12        2011
3               12        2011
4               12        2011
...            ...         ...
39712            7        2007
39713            7        2007
39714            7        2007
39715            7        2007
39716            6        2007

[39717 rows x 2 columns]

In [245]:
# check if loan_status has any empty values
loan_data.loan_status.isna().sum()

0

Since pymnt_plan has only one value, it's not required for analysis and hence can be dropped

In [246]:
# url is not helpful with analysis, hence dropping it
loan_data.drop('url', axis='columns', inplace=True)

In [247]:
# desc isn't helpful, dropping the column
loan_data.drop('desc', axis='columns', inplace=True)

In [248]:
loan_data.select_dtypes('object')

grade sub_grade                   emp_title emp_length home_ownership  \
0         B        B2                         NaN  10+ years           RENT   
1         C        C4                       Ryder   < 1 year           RENT   
2         C        C5                         NaN  10+ years           RENT   
3         C        C1         AIR RESOURCES BOARD  10+ years           RENT   
4         B        B5    University Medical Group     1 year           RENT   
...     ...       ...                         ...        ...            ...   
39712     A        A4             FiSite Research    4 years       MORTGAGE   
39713     C        C1  Squarewave Solutions, Ltd.    3 years           RENT   
39714     A        A4                         NaN   < 1 year       MORTGAGE   
39715     A        A2                         NaN   < 1 year       MORTGAGE   
39716     E        E2            Evergreen Center   < 1 year            OWN   

      verification_status issue_d  loan_status pymnt_plan             purpose  \
0                Verified  Dec-11   Fully Paid          n         credit_card   
1         Source Verified  Dec-11  Charged Off          n                 car   
2            Not Verified  Dec-11   Fully Paid          n      small_business   
3         Source Verified  Dec-11   Fully Paid          n               other   
4         Source Verified  Dec-11      Current          n               other   
...                   ...     ...          ...        ...                 ...   
39712        Not Verified  Jul-07   Fully Paid          n    home_improvement   
39713        Not Verified  Jul-07   Fully Paid          n         credit_card   
39714        Not Verified  Jul-07   Fully Paid          n  debt_consolidation   
39715        Not Verified  Jul-07   Fully Paid          n               other   
39716        Not Verified  Jun-07   Fully Paid          n  debt_consolidation   

                           title zip_code addr_state earliest_cr_line  \
0                       Computer    860xx         AZ           Jan-85   
1                           bike    309xx         GA           Apr-99   
2           real estate business    606xx         IL           Nov-01   
3                       personel    917xx         CA           Feb-96   
4                       Personal    972xx         OR           Jan-96   
...                          ...      ...        ...              ...   
39712           Home Improvement    802xx         CO           Nov-90   
39713  Retiring credit card debt    274xx         NC           Dec-86   
39714     MBA Loan Consolidation    017xx         MA           Oct-98   
39715                   JAL Loan    208xx         MD           Nov-88   
39716         Consolidation Loan    027xx         MA           Oct-03   

      revol_util initial_list_status last_pymnt_d next_pymnt_d  \
0         83.70%                   f       Jan-15          NaN   
1          9.40%                   f       Apr-13          NaN   
2         98.50%                   f       Jun-14          NaN   
3            21%                   f       Jan-15          NaN   
4         53.90%                   f       May-16       Jun-16   
...          ...                 ...          ...          ...   
39712     13.10%                   f       Jul-10          NaN   
39713     26.90%                   f       Jul-10          NaN   
39714     19.40%                   f       Apr-08          NaN   
39715      0.70%                   f       Jan-08          NaN   
39716     51.50%                   f       Jun-10          NaN   

      last_credit_pull_d application_type  
0                 May-16       INDIVIDUAL  
1                 Sep-13       INDIVIDUAL  
2                 May-16       INDIVIDUAL  
3                 Apr-16       INDIVIDUAL  
4                 May-16       INDIVIDUAL  
...                  ...              ...  
39712             Jun-10       INDIVIDUAL  
39713             Jul-10       INDIVIDUAL  
39714             

In [249]:
# find all columns that have only one value and remove them
column_unique = loan_data.nunique()
c1 = column_unique[column_unique.values == 1]
print(c1.index)

loan_data.drop(columns = list(c1.index), inplace=True)

Index(['pymnt_plan', 'initial_list_status', 'collections_12_mths_ex_med',
       'policy_code', 'application_type', 'acc_now_delinq',
       'chargeoff_within_12_mths', 'delinq_amnt', 'tax_liens'],
      dtype='object')


In [251]:
loan_data.select_dtypes('object')

grade sub_grade                   emp_title emp_length home_ownership  \
0         B        B2                         NaN  10+ years           RENT   
1         C        C4                       Ryder   < 1 year           RENT   
2         C        C5                         NaN  10+ years           RENT   
3         C        C1         AIR RESOURCES BOARD  10+ years           RENT   
4         B        B5    University Medical Group     1 year           RENT   
...     ...       ...                         ...        ...            ...   
39712     A        A4             FiSite Research    4 years       MORTGAGE   
39713     C        C1  Squarewave Solutions, Ltd.    3 years           RENT   
39714     A        A4                         NaN   < 1 year       MORTGAGE   
39715     A        A2                         NaN   < 1 year       MORTGAGE   
39716     E        E2            Evergreen Center   < 1 year            OWN   

      verification_status issue_d  loan_status             purpose  \
0                Verified  Dec-11   Fully Paid         credit_card   
1         Source Verified  Dec-11  Charged Off                 car   
2            Not Verified  Dec-11   Fully Paid      small_business   
3         Source Verified  Dec-11   Fully Paid               other   
4         Source Verified  Dec-11      Current               other   
...                   ...     ...          ...                 ...   
39712        Not Verified  Jul-07   Fully Paid    home_improvement   
39713        Not Verified  Jul-07   Fully Paid         credit_card   
39714        Not Verified  Jul-07   Fully Paid  debt_consolidation   
39715        Not Verified  Jul-07   Fully Paid               other   
39716        Not Verified  Jun-07   Fully Paid  debt_consolidation   

                           title zip_code addr_state earliest_cr_line  \
0                       Computer    860xx         AZ           Jan-85   
1                           bike    309xx         GA           Apr-99   
2           real estate business    606xx         IL           Nov-01   
3                       personel    917xx         CA           Feb-96   
4                       Personal    972xx         OR           Jan-96   
...                          ...      ...        ...              ...   
39712           Home Improvement    802xx         CO           Nov-90   
39713  Retiring credit card debt    274xx         NC           Dec-86   
39714     MBA Loan Consolidation    017xx         MA           Oct-98   
39715                   JAL Loan    208xx         MD           Nov-88   
39716         Consolidation Loan    027xx         MA           Oct-03   

      revol_util last_pymnt_d next_pymnt_d last_credit_pull_d  
0         83.70%       Jan-15          NaN             May-16  
1          9.40%       Apr-13          NaN             Sep-13  
2         98.50%       Jun-14          NaN             May-16  
3            21%       Jan-15          NaN             Apr-16  
4         53.90%       May-16       Jun-16             May-16  
...          ...          ...          ...                ...  
39712     13.10%       Jul-10          NaN             Jun-10  
39713     26.90%       Jul-10          NaN             Jul-10  
39714     19.40%       Apr-08          NaN             Jun-07  
39715      0.70%       Jan-08          NaN             Jun-07  
39716     51.50%       Jun-10          NaN             Jun-10  

[39717 rows x 17 columns]

In [252]:
loan_data.head(5)

id  member_id  loan_amnt  funded_amnt  funded_amnt_inv  term  \
0  1077501    1296599       5000         5000           4975.0    36   
1  1077430    1314167       2500         2500           2500.0    60   
2  1077175    1313524       2400         2400           2400.0    36   
3  1076863    1277178      10000        10000          10000.0    36   
4  1075358    1311748       3000         3000           3000.0    60   

   int_rate  installment grade sub_grade  ... total_rec_late_fee recoveries  \
0     10.65       162.87     B        B2  ...               0.00       0.00   
1     15.27        59.83     C        C4  ...               0.00     117.08   
2     15.96        84.33     C        C5  ...               0.00       0.00   
3     13.49       339.31     C        C1  ...              16.97       0.00   
4     12.69        67.79     B        B5  ...               0.00       0.00   

  collection_recovery_fee  last_pymnt_d last_pymnt_amnt next_pymnt_d  \
0                    0.00        Jan-15          171.62          NaN   
1                    1.11        Apr-13          119.66          NaN   
2                    0.00        Jun-14          649.91          NaN   
3                    0.00        Jan-15          357.48          NaN   
4                    0.00        May-16           67.79       Jun-16   

  last_credit_pull_d pub_rec_bankruptcies issue_month issue_year  
0             May-16                  0.0          12       2011  
1             Sep-13                  0.0          12       2011  
2             May-16                  0.0          12       2011  
3             Apr-16                  0.0          12       2011  
4             May-16                  0.0          12       2011  

[5 rows x 48 columns]